packages/sit-onyx/src/components/OnyxNavItem/OnyxNavItem.vue

```diff
@@ -0,0 +1,124 @@
+<script lang="ts" setup>
+import { computed, ref } from "vue";
+import type { OnyxNavItemProps } from "./types";
+import OnyxListbox from "../OnyxListbox/OnyxListbox.vue";
+import type { ListboxOption } from "../OnyxListbox/types";
+
+const props = withDefaults(defineProps<OnyxNavItemProps>(), {
+  active: false,
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the navItem is clicked (via click or keyboard).
+   */
+  navigate: [href: string];
+}>();
+
+defineSlots<{
+  /**
+   * A default slot that can be used for OnyxBadge.
+   */
+  default(): unknown;
+}>();
+
+const listboxVisible = ref(false);
+
+const nestedOptions = computed(() => {
+  return props.options?.map((opt) => {
+    return { id: opt.href, label: opt.label };
+  }) satisfies ListboxOption[] | undefined;
+});
+
+function hideListbox() {
```

Comment:
Would prefer to do this via CSS and `transition`.
If implementing it via plain CSS doesn't work, you can use `v-show` instead of `v-if` which is set `display: none` so we can animate it with CSS